cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactTable.java

```diff
@@ -130,12 +130,22 @@ public void add(List<Fact> facts) {
   }
 
   public FactScanner scan(FactScan scan) {
-    return new FactScanner(getScanner(scan), codec, scan.getStartTs(), scan.getEndTs());
+    return new FactScanner(getScanner(scan), codec, scan.getStartTs(), scan.getEndTs(), scan.getMeasureNames());
   }
 
   private Scanner getScanner(FactScan scan) {
-    byte[] startRow = codec.createStartRowKey(scan.getTagValues(), scan.getMeasureName(), scan.getStartTs(), false);
-    byte[] endRow = codec.createEndRowKey(scan.getTagValues(), scan.getMeasureName(), scan.getEndTs(), false);
+    String firstMeasureName = null;
+    String lastMeasureName = null;
+
+    if (scan.getMeasureNames() != null && !scan.getMeasureNames().isEmpty()) {
+      List<String> measures = Lists.newArrayList(scan.getMeasureNames());
+      Collections.sort(measures);
```

Comment:
hm hm. This is not correct... Comparing strings is not same as comparing bytes lexographically (rows order in table)..

I think it is fine to pass null, if you are querying for more than one metric name for sake of simplicity of code. You won't get a lot out of this optimization